+Prefs/+Inputs/ButtonField.m

```diff
@@ -13,23 +9,13 @@
         function [obj,height_px,label_width_px] = make_UI(obj,pref,parent,yloc_px,width_px,margin)
             % Here, widths will all be taken care of in adjust_UI
             [obj,height_px,label_width_px] = make_UI@Prefs.Inputs.LabelControlBasic(obj,pref,parent,yloc_px,width_px);
-            set(obj.ui,'TooltipString',pref.value);
-            if isempty(pref.value)
-                set(obj.ui,'String',obj.empty_string);
-            end
+            set(obj.ui,'String',pref.string);
         end
         function set_value(obj,val)
-            if isempty(val)
-                obj.ui.String = obj.empty_string;
-            else
-                [~,name,ext] = fileparts(val);
-                obj.ui.String = [name, ext];
-            end
-            obj.ui.UserData.value = val;
-            obj.ui.TooltipString = val;
+            % Do nothing.
         end
         function val = get_value(obj)
-            val = obj.ui.UserData.value;
+            val = false;
```

Comment:
Looks like you may have repurposed the UI's UserData, but just curious to thought process here. You don't want users to be able to pass data back from a button?

Reply:
I just want the button press to execute a function. Maybe the value should be whether execution was successful?